apps/api/src/search/searxng.ts

```diff
@@ -0,0 +1,66 @@
+import axios from "axios";
+import dotenv from "dotenv";
+import { SearchResult } from "../../src/lib/entities";
+
+dotenv.config();
+
+interface SearchOptions {
+  tbs?: string;
+  filter?: string;
+  lang?: string;
+  country?: string;
+  location?: string;
+  num_results: number;
+  page?: number;
+}
+
+export async function searxng_search(
+  q: string,
+  options: SearchOptions,
+): Promise<SearchResult[]> {
+  const params = {
+    q: q,
+    language: options.lang,
+    // gl: options.country, //not possible with SearXNG
+    // location: options.location, //not possible with SearXNG
+    // num: options.num_results, //not possible with SearXNG
+    engines: process.env.SEARXNG_ENGINES || "",
+    categories: process.env.SEARXNG_CATEGORIES || "general",
+    pageno: options.page ?? 1,
+    format: "json"
+  };
+
+  const url = process.env.SEARXNG_ENDPOINT as string;
+  if (!url) {
+    console.error(`SEARXNG_ENDPOINT environment variable is not set`);
+  }
```

Comment:
Not required since it's only called if the endpoint is defined
```suggestion
  const url = process.env.SEARXNG_ENDPOINT!;
```